README.md

```diff
@@ -21,30 +21,65 @@ There are 4 layers within the application:
 </p>
 
 ## Overview
-In order to understand how all this work together, it's important to talk about each component role inside the Cpresentation layer.
+In order to understand how this work together, it's important to talk about each component's role inside the presentation layer.
 
-First of all, the architecture. It's a combination of the MVI and Redux patterns.
-The MVI patter has three main components: Intent, Model, and View.
-The intent refers to the intention of change the state of the app, so in Gong case it would be the actions, delivered then to viewModel.
-ViewModel represents the model component of the pattern. It is responsible of creation of a new state, which is an immutable data structure.
-At any given moment, there is just one state in the app, which represent a single source of truth.
+First of all, the architecture. It's a combination of the MVI and Redux architecture patterns.
+The MVI pattern has three main components: Intent, Model, and View.
+The intent refers to the intention to change the state of the app, so in Gong's case, it would be the actions, delivered then to ViewModel.
+ViewModel holds the model component of the pattern. It is responsible of creation of a new state, which is an immutable data structure.
+At any given moment, there is only one state in the app, which represents a single source of truth.
 The only way to change the state is to create a new one, triggered by the actions. But when and how is the new state created?
 The Redux pattern comes up at this point. Redux is a pattern and library for managing and updating application state, using events called "actions".
 More precisely, the main components of Redux are State, Action, and Reducer.
 In Gong the composables communicate actions to the viewModels so they can manage and emit the state back to view.
 This ensure state can only be updated in a predictable way.
 Then, inside the model, the reducer is called with a proper action and the latest state and forward its result as an output value of the model.
-Reducer is a function that takes the previous state and action and creates a new state, and in Gong this role is played by the function ‘processAction’, located in viewModels.
+Reducer is a function that takes the previous state and action and creates a new state, and in Gong this role is played by the function `processAction`, located in viewModels.
 
 To continue with the insight of the project, let's see how this is done.
 With the shared flow, actions are broadcast to an unknown number (zero or more) of subscribers.
 In the absence of a subscriber, any posted action is immediately dropped. It is a design pattern to use for actions that must be processed immediately or not at all.
-The viewModel handles each action in the ‘processAction’ method. Whenever an action is added to the "contract", it also has to be added here.
+
+<p align="center">
+  <img src="/images/SharedFlow.png">
```

Comment:
what do you think of adding something more representative to display the flow? I don't know, the user clicks something and generates an action it's processed and.... maybe we can discuss it